### ISSUE TITLE
[Core]db_filename only accepts absolute paths [5.0.0-a37f798]

### ISSUE THREAD

**Issue by maxpowa** [embolalia/willie#714]
It appears to be unable to create the file if you set db_filename in [core], or does not properly access it. Un-setting the value causes the default.db database to be created (and used) with no issues.

Using latest git, a37f798

```
Critical exception in coreTraceback (most recent call last):
  File "/home/willie/Snakepit/local/lib/python2.7/site-packages/willie/__init__.py", line 50, in run
    p = bot.Willie(config)
  File "/home/willie/Snakepit/local/lib/python2.7/site-packages/willie/bot.py", line 91, in __init__
    self.db = WillieDB(config)
  File "/home/willie/Snakepit/local/lib/python2.7/site-packages/willie/db.py", line 42, in __init__
    self._create()
  File "/home/willie/Snakepit/local/lib/python2.7/site-packages/willie/db.py", line 71, in _create
    'CREATE TABLE nick_ids (nick_id INTEGER PRIMARY KEY AUTOINCREMENT)'
  File "/home/willie/Snakepit/local/lib/python2.7/site-packages/willie/db.py", line 53, in execute
    with self.connect() as conn:
  File "/home/willie/Snakepit/local/lib/python2.7/site-packages/willie/db.py", line 46, in connect
    return sqlite3.connect(self.filename)
OperationalError: unable to open database file
----------------------------------------
```

**Comment by tyrope** [comment]
I have set db_filename in 2 bots that run just fine. Did you set it to be an absolute path? What OS are you using?

**Comment by maxpowa** [comment]
Ubuntu Server 14.04. I wasn't using an absolute path, I was using a home directory path. ```~/.willie/database.db``` Never had any issues using a home dir path with the old setting.

**Comment by tyrope** [comment]
I encountered the same issue. it appears we're no longer translating, please change to /home/%USERNAME%/.willie/database.db and things should work as expected.

**Comment by embolalia** [comment]
Fixed in 9ac8831210db3fa0e3592c7388ff15f62e42309d; you can now use `~` in the db filename.
